lib/prime.rb

```diff
@@ -49,6 +76,28 @@ def prime?
   def Integer.each_prime(ubound, &block) # :yields: prime
     Prime.each(ubound, &block)
   end
+
+  private
+  def sufficient_miller_rabin_bases
+    # Miller-Rabin's complexity is O(k log^3n).
+    # So we can reduce the complexity by reducing the number of bases tested.
+    # Using values from https://en.wikipedia.org/wiki/Miller%E2%80%93Rabin_primality_test
+    return [2]                                  if self < 2_047
+    return [2,3]                                if self < 1_373_653
+    return [31,73]                              if self < 9_080_191
+    return [2,3,5]                              if self < 25_326_001
+    return [2,3,5,7]                            if self < 3_215_031_751
+    return [2,7,61]                             if self < 4_759_123_141
+    return [2,13,23,1662803]                    if self < 1_122_004_669_633
+    return [2,3,5,7,11]                         if self < 2_152_302_898_747
+    return [2,3,5,7,11,13]                      if self < 3_474_749_660_383
+    return [2,3,5,7,11,13,17]                   if self < 341_550_071_728_321
+    return [2,3,5,7,11,13,17,19,23]             if self < 3_825_123_056_546_413_051
+    return [2,3,5,7,11,13,17,19,23,29,31,37]    if self < 318_665_857_834_031_151_167_461
+    return [2,3,5,7,11,13,17,19,23,29,31,37,41] if self < 3_317_044_064_679_887_385_961_981
```

Comment:
Is there a reason to test the last case for this private method?
If all conditions fail the method will return ``nil``, which may or may not be what you want.

Reply:
I think i'll just leave this as if we some day have new best-knowns, it would be wanted, and as you noticed, it only ever happens when N is absolutely galactic in size....